bot/cogs/information.py

```diff
@@ -211,53 +223,85 @@ async def create_user_embed(self, ctx: Context, user: Member) -> Embed:
         # Custom status
         custom_status = ''
         for activity in user.activities:
-            # Check activity.state for None value if user has a custom status set
-            # This guards against a custom status with an emoji but no text, which will cause
-            # escape_markdown to raise an exception
-            # This can be reworked after a move to d.py 1.3.0+, which adds a CustomActivity class
-            if activity.name == 'Custom Status' and activity.state:
-                state = escape_markdown(activity.state)
-                custom_status = f'Status: {state}\n'
+            if isinstance(activity, CustomActivity):
+                state = ""
+
+                if activity.name:
+                    state = escape_markdown(activity.name)
+
+                emoji = ""
+                if activity.emoji:
+                    if not activity.emoji.id:
+                        emoji += activity.emoji.name + " "
+
+                custom_status = f'Status: {emoji}{state}\n'
 
         name = str(user)
         if user.nick:
             name = f"{user.nick} ({name})"
 
+        badges = ""
+
+        for badge, is_set in user.public_flags:
+            if is_set and (emoji := getattr(constants.Emojis, f"badge_{badge}")):
+                badges += emoji + " "
```

Comment:
Neat way to use walrus. But I think you want to specify a default for `getattr`. [This way can raise](https://docs.python.org/3/library/functions.html#getattr).

I'd probably personally make `badges` a list & then join the elements on a `" "`, but this is fine too.